colossalai/inference/tensor_parallel/modeling/llama.py

```diff
@@ -294,15 +301,26 @@ def llama_flash_attn_kvcache_forward(
 
             attn_output = torch.empty_like(query_states)
 
-            llama_context_attn_fwd(
-                query_states,
-                key_states,
-                value_states,
-                attn_output,
-                infer_state.start_loc,
-                infer_state.seq_len,
-                infer_state.cache_manager.past_key_values_length,
-            )
+            if self.num_key_value_groups == 1:
```

Comment:
self.num_key_value_groups!=1 means using group attention?

Reply:
self.num_key_value_groups = self.num_heads // self.num_key_value_heads is defined in transformers LlamaAttention. This means self.num_key_value_groups !=1 uses group attention.